### PR TITLE
raftstore-v2: dynamic change store pool size

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Ref #14485

What's Changed:

```commit-message
implement dynamic change store pool size
```

Manual test:
image

todo:
support change other pools such as async-write threads and apply threads.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/batch-system/src/batch.rs [hunk 1]
```diff
@@ -466,6 +474,11 @@ impl<N: Fsm, C: Fsm, Handler: PollHandler<N, C>> Poller<N, C, Handler> {
             }
         }
         batch.clear();
+
+        info!(
```
Comment (tonyxuqqi) [inline components/batch-system/src/batch.rs:478]: What's the purpose of this log. Same as the one above.

**Review by tonyxuqqi — APPROVED** [review]

File: components/batch-system/src/batch.rs (hunk 1 above)
Comment (SpadeA-Tang) [reply components/batch-system/src/batch.rs:478]: It's used for me to debug. I will remove them later.

**Review by bufferflies — APPROVED** [review]

**Pull request merged**